unstructured_ingest/v2/processes/connectors/sql/vastdb.py

```diff
@@ -0,0 +1,281 @@
+        # but also allow for additional columns
+        approved_columns = set(list(_COLUMNS) + self.upload_stager_config.additional_columns)
+        element = {k: v for k, v in data.items() if k in approved_columns}
+        element[RECORD_ID_LABEL] = file_data.identifier
+        return element
+
+    def conform_dataframe(self, df: pd.DataFrame) -> pd.DataFrame:
+        for column in filter(lambda x: x in df.columns, _DATE_COLUMNS):
+            df[column] = df[column].apply(parse_date_string).apply(lambda date: date.timestamp())
+        for column in filter(
+            lambda x: x in df.columns,
+            ("permissions_data", "record_locator", "points", "links"),
+        ):
+            df[column] = df[column].apply(
+                lambda x: json.dumps(x) if isinstance(x, (list, dict)) else None
+            )
+        for column in filter(
+            lambda x: x in df.columns,
+            ("version", "page_number", "regex_metadata"),
+        ):
+            df[column] = df[column].apply(str)
+        if self.upload_stager_config.rename_columns_map:
+            df.rename(columns=self.upload_stager_config.rename_columns_map, inplace=True)
+        return df
+
+
+class VastdbUploaderConfig(SQLUploaderConfig):
+    pass
+
+
+@dataclass
+class VastdbUploader(SQLUploader):
+    upload_config: VastdbUploaderConfig = field(default_factory=VastdbUploaderConfig)
+    connection_config: VastdbConnectionConfig
+    connector_type: str = CONNECTOR_TYPE
+
+    def precheck(self) -> None:
+        try:
+            with self.get_cursor() as cursor:
+                table = self.connection_config.get_table(cursor, self.upload_config.table_name)
+                table.select()
+        except Exception as e:
+            logger.error(f"failed to validate connection: {e}", exc_info=True)
+            raise DestinationConnectionError(f"failed to validate connection: {e}")
+
+    @requires_dependencies(["vastdb", "ibis", "pyarrow"], extras="vastdb")
+    def upload_dataframe(self, df: pd.DataFrame, file_data: FileData) -> None:
+        import pyarrow as pa
+
+        if self.can_delete():
+            self.delete_by_record_id(file_data=file_data)
+        else:
+            logger.warning(
+                f"table doesn't contain expected "
+                f"record id column "
+                f"{self.upload_config.record_id_key}, skipping delete"
+            )
+        df.replace({np.nan: None}, inplace=True)
+        df = self._fit_to_schema(df=df)
+
+        logger.info(
+            f"writing a total of {len(df)} elements via"
+            f" document batches to destination"
+            f" table named {self.upload_config.table_name}"
+            f" with batch size {self.upload_config.batch_size}"
+        )
+
+        for rows in split_dataframe(df=df, chunk_size=self.upload_config.batch_size):
+
+            with self.get_cursor() as cursor:
+                pa_table = pa.Table.from_pandas(df)
```

Comment:
I assume this needs to be rows, otherwise you're uploading the entire table for each batch. 
```suggestion
                pa_table = pa.Table.from_pandas(rows)
```

Reply:
thank you!